File: src/entities/product.ts
import { IDRef } from "./types";

export class Product {
  productId!: IDRef;
  sku!: string;
  category!: string;
  name!: string;
  brand!: string;
  variant!: string;
  quantity!: number;
  price!: number;
  currency!: string;
  imageUri!: string;
  uri!: string;
}

File: event-schemas/data/golang/Product.go

package entities

// Product represents a Product model.
type Product struct {
  ProductId string
  Sku string
  Category string
  Name string
  Brand string
  Variant string
  Quantity float64
  Price float64
  Currency string
  ImageUri string
  Uri string
  AdditionalProperties map[string][]interface{}
}